hack/pre-run-e2e.sh

```diff
@@ -0,0 +1,266 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+if [[ "${REGISTRY_USER_NAME}" = "" ]] || [[ "${REGISTRY_PASSWORD}" = "" ]]; then
+  echo "REGISTRY_USER_NAME and REGISTRY_PASSWORD can not be empty, please set it"
+  exit 1
+fi
+
+export REGISTRY_SERVER_ADDRESS=swr.ap-southeast-1.myhuaweicloud.com
+export REGISTRY=swr.ap-southeast-1.myhuaweicloud.com/dev-registry
+export VERSION=latest
+
+echo -e "\nBuild images"
+make upload-images
```

Comment:
OK, I see, I changed `dev-registry/huawei-cloud-controller-manager` to a private image.
And use `git rev-parse --short HEAD` to generate the image version.

Reply:
Actually, I think we don't need to push the image to the global registry. The ideal way is to build the image locally and `load` it to the target cluster running the CCM.

In addition, consider a such scenario:
A contributor probably wants to run the E2E test locally before sending a PR, that's pretty reasonable, but he/she must have the registry certificate first.  It doesn't make sense, right?

Reply:
> A contributor probably wants to run the E2E test locally before sending a PR, that's pretty reasonable, but he/she must  have the registry certificate first. It doesn't make sense, right?

yeah, this will discourage other contributors from running e2e test.

Reply:
> Actually, I think we don't need to push the image to the global registry. The ideal way is to build the image locally and load it to the target cluster running the CCM.

Great idea, since the current way suffices for CI testing, I'll do the optimization in another PR.